File: java/scf-simple/src/main/java/org/mydotey/scf/source/stringproperty/StringPropertyConfigurationSource.java
package org.mydotey.scf.source.stringproperty;

import org.mydotey.scf.AbstractConfigurationSource;
import org.mydotey.scf.ConfigurationSourceConfig;

/**
 * @author koqizhao
 *
 * May 17, 2018
 */
public abstract class StringPropertyConfigurationSource<C extends ConfigurationSourceConfig>
        extends AbstractConfigurationSource<C> {

    public StringPropertyConfigurationSource(C config) {
        super(config);
    }

    @Override
    public Object getPropertyValue(Object key) {
        if (key.getClass() != String.class)
            return null;

        return getPropertyValue((String) key);
    }

    public abstract String getPropertyValue(String key);

}

File: dotnet/src/MyDotey.SCF.Simple/Source/StringProperty/StringPropertyConfigurationSource.cs
using System;

namespace MyDotey.SCF.Source.StringProperty
{
    /**
     * @author koqizhao
     *
     * May 17, 2018
     */
    public abstract class StringPropertyConfigurationSource<C> : AbstractConfigurationSource<C>
        where C : ConfigurationSourceConfig
    {
        public StringPropertyConfigurationSource(C config)
            : base(config)
        {
        }

        protected override object getPropertyValue(object key)
        {
            if (!(key is string))
                return null;

            return getPropertyValue((string)key);
        }

        public abstract string getPropertyValue(string key);
    }
}